### PR TITLE
i18n(es): Translated recipe `build-forms.mdx`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### What kind of changes does this PR include?

- Translated content

#### Description

- What does this PR change? Give us a brief description.
Translated recipe `build-forms.mdx` to spanish

### CODE REVIEW

File: src/content/docs/es/recipes/build-forms.mdx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+- Un proyecto con [SSR](/es/guides/server-side-rendering/) (`output: 'server'`) habilitado
+
+## Receta
+
+1. Crea o identifica una página `.astro` que contenga tu formulario y tu código de manejo. Por ejemplo, podrías agregar una página de registro:
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:21]:
```suggestion
    <h1>Registro</h1>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    Asegúrate de incluir un elemento `<button>` o `<input type="submit">` para enviar el formulario.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:34]:
```suggestion
        Nombre de usuario:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="text" name="username" />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" />
+      </label>
+      <label>
+        Password:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:42]:
```suggestion
        Contraseña:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="password" name="password" />
+      </label>
+      <button>Submit</button>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:45]:
```suggestion
      <button>Enviar</button>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    - `minlength` establece una longitud mínima requerida para el texto de entrada.
+    - `type="email"` también introduce validación que solo aceptará un formato de correo electrónico válido.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:62]:
```suggestion
        Nombre de usuario:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="text" name="username" required />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      <label>
+        Password:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:70]:
```suggestion
        Contraseña:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:73]:
```suggestion
      <button>Enviar</button>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+4. El envío del formulario causará que el navegador solicite la página nuevamente. Cambia el método de transferencia de datos del formulario a `POST` para enviar los datos del formulario como parte del cuerpo de la `Request`, en lugar de como parámetros de URL.
+
+    ```astro title="src/pages/register.astro" 'method="POST"'
+    ---
+    ---
+    <h1>Register</h1>
+    <form method="POST">
+      <label>
+        Username:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:91]:
```suggestion
        Nombre de usuario:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="text" name="username" required />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      <label>
+        Password:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:99]:
```suggestion
        Contraseña:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:102]:
```suggestion
      <button>Enviar</button>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+
+    Para escribir validación de lógica compleja más fácilmente, puedes construir tu formulario usando un [framework de frontend](/es/core-concepts/framework-components/) y elegir una librería de formularios como [React Hook Form](https://react-hook-form.com/) o [Felte](https://felte.dev/).
+    :::
+
+4. El envío del formulario causará que el navegador solicite la página nuevamente. Cambia el método de transferencia de datos del formulario a `POST` para enviar los datos del formulario como parte del cuerpo de la `Request`, en lugar de como parámetros de URL.
+
+    ```astro title="src/pages/register.astro" 'method="POST"'
+    ---
+    ---
+    <h1>Register</h1>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:88]:
```suggestion
    <h1>Registro</h1>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+          return Astro.redirect("/login");
+        }
+      } catch (error) {
+        if (error instanceof Error) {
+          console.error(error.message);
+        }
+      }
+    }
+    ---
+    <h1>Register</h1>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:182]:
```suggestion
    <h1>Registro</h1>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+    <form method="POST">
+      <label>
+        Username:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:185]:
```suggestion
        Nombre de usuario:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="text" name="username" />
+      </label>
+      {errors.username && <p>{errors.username}</p>}
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      {errors.email && <p>{errors.email}</p>}
+      <label>
+        Password:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:195]:
```suggestion
        Contraseña:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      {errors.password && <p>{errors.password}</p>}
+      <button>Register</button>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:199]:
```suggestion
      <button>Registro</button>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+        const password = data.get("password");
+        // Do something with the data
+      } catch (error) {
+        if (error instanceof Error) {
+          console.error(error.message);
+        }
+      }
+    }
+    ---
+    <h1>Register</h1>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:124]:
```suggestion
    <h1>Registro</h1>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+    <form method="POST">
+      <label>
+        Username:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:127]:
```suggestion
        Nombre de usuario:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="text" name="username" required />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      <label>
+        Password:
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:135]:
```suggestion
        Contraseña:
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 19]
```diff
@@ -0,0 +1,204 @@
...
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:138]:
```suggestion
      <button>Enviar</button>
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 20]
```diff
@@ -0,0 +1,204 @@
+---
+title: Construye formularios HTML en páginas Astro
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:2]:
```suggestion
title: Construye formularios HTML en páginas de Astro
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 21]
```diff
@@ -0,0 +1,204 @@
...
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+
+5.  Comprueba el método `POST` en el frontmatter y accede a los datos del formulario usando `Astro.request.formData()`. Envuelve esto en un bloque `try ... catch` para manejar los casos en los que la solicitud `POST` no fue enviada por un formulario y el ``formData es inválido.
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:106]:
```suggestion
5.  Comprueba el método `POST` en el frontmatter y accede a los datos del formulario usando `Astro.request.formData()`. Envuelve esto en un bloque `try ... catch` para manejar los casos en los que la solicitud `POST` no fue enviada por un formulario y el `formData` es inválido.
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+
+5. Valida los datos del formulario en el servidor. Esto debería incluir la misma validación realizada en el cliente para evitar envíos maliciosos a tu punto final y para admitir el raro navegador heredado que no tiene validación de formularios.
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:142]:
```suggestion
5. Valida los datos del formulario en el servidor. Esto debería incluir la misma validación realizada en el cliente para evitar envíos maliciosos a tu endpoint y brindar soporte a los raros navegadores heredados que no tienen validación de formularios.
```

File: src/content/docs/es/recipes/build-forms.mdx [hunk 23]
```diff
@@ -0,0 +1,204 @@
...
+
+    También puede incluir validación que no se puede hacer en el cliente. Por ejemplo, este ejemplo verifica si el correo electrónico ya está en la base de datos.
```
Comment (Waxer59) [inline src/content/docs/es/recipes/build-forms.mdx:144]:
```suggestion
    También puedes incluir validación que no se puede hacer en el cliente. Por ejemplo, este ejemplo verifica si el correo electrónico ya está en la base de datos.
```

**Review by Waxer59 — COMMENTED** [review]
Buena PR 🤠, te dejo unas sugerencias

**Comment by dreyfus92** [comment]
Muchas gracias por la revisión @Waxer59 😁

**Review by Waxer59 — APPROVED** [review]
LGTM ✔

**Pull request merged**